src/ParamMap.h

```diff
@@ -0,0 +1,128 @@
+#ifndef HALIDE_PARAM_MAP_H
+#define HALIDE_PARAM_MAP_H
+
+/** \file
+ * Defines a collection of parameters to be passed as formal arugments
+ * to a JIT invocation.
+ */
+
+#include "Param.h"
+#include "Parameter.h"
+
+namespace Halide {
+
+class ImageParam;
+
+class ParamMap {
```

Comment:
A good chunk of this .h file could be moved into the .cpp file (and probably should be).